src/labeler.ts

```diff
@@ -122,110 +115,178 @@ async function fetchContent(
+    core.debug(` checking config ${JSON.stringify(config)}`);
+    if (!checkMatch(changedFiles, config)) {
+      return false;
     }
   }
-  return false;
+
+  return true;
 }
 
-function isMatch(changedFile: string, matchers: Minimatch[]): boolean {
-  core.debug(`    matching patterns against file ${changedFile}`);
-  for (const matcher of matchers) {
-    core.debug(`   - ${printPattern(matcher)}`);
-    if (!matcher.match(changedFile)) {
-      core.debug(`   ${printPattern(matcher)} did not match`);
+function checkMatch(changedFiles: string[], matchConfig: MatchConfig): boolean {
+  if (!Object.keys(matchConfig).length) {
+    core.debug(`  no "any" or "all" patterns to check`);
+    return false;
+  }
+
+  if (matchConfig.all) {
+    if (!checkAll(matchConfig.all, changedFiles)) {
+      return false;
+    }
+  }
+
+  if (matchConfig.any) {
+    if (!checkAny(matchConfig.any, changedFiles)) {
       return false;
     }
   }
 
-  core.debug(`   all patterns matched`);
   return true;
 }
 
 // equivalent to "Array.some()" but expanded for debugging and clarity
-function checkAny(changedFiles: string[], globs: string[]): boolean {
-  const matchers = globs.map(g => new Minimatch(g));
+export function checkAny(
+  matchConfigs: BaseMatchConfig[],
+  changedFiles: string[]
+): boolean {
   core.debug(`  checking "any" patterns`);
-  for (const changedFile of changedFiles) {
-    if (isMatch(changedFile, matchers)) {
-      core.debug(`  "any" patterns matched against ${changedFile}`);
-      return true;
+  if (!Object.keys(matchConfigs).length) {
+    core.debug(`  no "any" patterns to check`);
+    return false;
+  }
+
+  for (const matchConfig of matchConfigs) {
+    if (matchConfig.baseBranch) {
+      if (checkAnyBranch(matchConfig.baseBranch, 'base')) {
+        return true;
+      }
+    }
+
+    if (matchConfig.changedFiles) {
+      if (checkAnyChangedFiles(changedFiles, matchConfig.changedFiles)) {
+        return true;
+      }
+    }
+
+    if (matchConfig.headBranch) {
+      if (checkAnyBranch(matchConfig.headBranch, 'head')) {
+        return true;
+      }
     }
   }
 
-  core.debug(`  "any" patterns did not match any files`);
+  core.debug(`  "any" patterns did not match any configs`);
   return false;
 }
 
 // equivalent to "Array.every()" but expanded for debugging and clarity
-function checkAll(changedFiles: string[], globs: string[]): boolean {
-  const matchers = globs.map(g => new Minimatch(g));
-  core.debug(` checking "all" patterns`);
-  for (const changedFile of changedFiles) {
-    if (!isMatch(changedFile, matchers)) {
-      core.debug(`  "all" patterns did not match against ${changedFile}`);
-      return false;
-    }
+export function checkAll(
+  matchConfigs: BaseMatchConfig[],
+  changedFiles: string[]
+): boolean {
+  core.debug(`  checking "all" patterns`);
+  if (!Object.keys(matchConfigs).length) {
+    core.debug(`  no "all" patterns to check`);
+    return false;
   }
```

Comment:
I guess this validation never works the way it should since `matchConfigs` is an array. The length of this array `[{}]` will be `1` and the `checkAll` function will return `true`

Reply:
Sorry for the vague description of the problem! I meant that the `toMatchConfig` function always returns an object so `matchConfigs` cannot be empty. The returned object can be empty if we specify invalid config, for example this
```yml
LabelName:
- all:
  - invalid-option: ['**']
```
will lead to `{"all":[{}]}` and length will be `1`.

The following update should fix the problem:
```js
  const ALLOWED_CONFIG_KEYS = ['changedFiles', 'headBranch', 'baseBranch'];

  if (!matchConfigs.some(configOption => ALLOWED_CONFIG_KEYS.includes(Object.keys(configOption)[0]))) {
    core.debug(`  no "all" patterns to check`);
    return false;
  }
```

Reply:
Ahh I see, thanks! I've updated this now 👍